src/Components/SuperAdmins/index.jsx

```diff
@@ -2,58 +2,145 @@ import { useEffect, useState } from 'react';
 import styles from './super-admins.module.css';
 import Table from './Table';
 import Modal from './Modal';
+import Form from './Form';
 
 const SuperAdmins = () => {
-  const [showModal, setShowModal] = useState(false);
+  const [showDeleteWarning, setShowDeleteWarning] = useState(false);
+  const [showSuperAdminForm, setShowSuperAdminForm] = useState(false);
+  const [method, setMethod] = useState('');
   const [superAdmins, setSuperAdmins] = useState([]);
   const [superAdminId, setSuperAdminId] = useState();
 
   const getSuperAdmins = async () => {
-    const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins`);
-    const data = await response.json();
-    setSuperAdmins(data.data);
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins`);
+      const data = await response.json();
+      setSuperAdmins(data.data);
+    } catch (error) {
+      console.error(error);
+      throw new Error('An error has ocurred, cannot get the Super Admins');
+    }
   };
 
+  useEffect(() => {
+    getSuperAdmins();
+  }, []);
+
   const deleteSuperAdmin = async (id) => {
     try {
-      await fetch(`${process.env.REACT_APP_API_URL}/super-admins/${id}`, {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins/${id}`, {
         method: 'DELETE'
       });
+      const data = await response.json();
+      alert(data.message);
     } catch (error) {
       console.error(error);
     }
   };
 
-  useEffect(() => {
-    getSuperAdmins();
-  }, []);
+  const addSuperAdmin = async (superAdmin) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins/`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(superAdmin)
+      });
+      const data = await response.json();
+      if (!response.ok) {
+        alert(data.message);
+      } else {
+        getSuperAdmins();
```

Comment:
Instead of calling the getSuperAdmins function after each action to update the list of super administrators, you can update the superAdmins status directly after a successful action.
```suggestion
setSuperAdmins([...superAdmins, data.data]);
```